bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaAdjustableSlatsRollerShutterHandler.java

```diff
@@ -87,7 +87,10 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             if (OnOffType.ON.equals(command)) {
                 sendCommand(COMMAND_SET_ROCKERPOSITION);
             }
-        } else {
+        } else if (CLOSURE_AND_ORIENTATION.equals(channelUID.getId())) {
```

Comment:
Add check for `!CLOSURE_AND_ORIENTATION.equals(...)` earlier (line 79f).

Otherwise does a different implementation logic is easier to maintain:

```java
      public void handleCommand(ChannelUID channelUID, Command command) {
        super.handleCommand(channelUID, command);	
        switch(channelUID.getId()) {
             case ROCKER:
                 // TODO
                 break;
             case CLOSURE_AND_ORIENTATION:
                 // TODO
                 break;
             ...
             default:
                 return;
        }
    }

Reply:
used the suggested implementation